File: PRACTICA1/JAVA/deepspace/src/deepspace/Damage.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package deepspace;

import java.util.ArrayList;

/**
 *
 * @author celia
 */
public abstract class Damage {
    private int nShields;
    
    
    
    public Damage(int s){
       nShields = s;
    }
    
    abstract public DamageToUI getUIversion();
    
    abstract public Damage adjust(ArrayList<Weapon> w,ArrayList<ShieldBooster> s);
   
    protected int adjustShields(ArrayList<ShieldBooster> s){
        int shields;
        if (getNShields() > s.size())
            shields = s.size();
        else 
            shields = getNShields();
        return shields;
    }
    public void discardShieldBooster(){
        if (nShields != 0)
            nShields --;
    }
    public boolean hasNoEffect(){
        return (nShields == 0);
    }
    public int getNShields(){
        return nShields;
    }
    
    public String toString(){
        String mensaje = "El daño es "+nShields+" escudos ";
        return mensaje;
    }
    abstract public void discardWeapon(Weapon w);
}

File: PRACTICA1/RUBY/Deepspace/Damage.rb
#encoding:utf-8

module Deepspace

require_relative "DamageToUI"

# @author Lucía Salamanca López, Celia Arias Martínez

class Damage

  def initialize(s)
    @nShields = s #int
  end

  def getUIversion
    DamageToUI.new(self)
  end

  
  def adjust(s)  #w es array de Weapons y s array de SHieldbooster
    if (nShields > s.length)
      shields = s.length
    else
      shields = nShields
    end
    return shields
  end
  

  def discardShieldBooster
    if (@nShields != 0)
      @nShields = @nShields-1
    end
  end

  def hasNoEffect
    return (nShields == 0)
  end

  def nShields
    @nShields
  end

  def to_s
   "El daño es #{@nShields} escudos, "
  end


end

end
